ganga/GangaCore/Utility/Shell.py

```diff
@@ -116,13 +116,11 @@ def __init__(self, setup=None, setup_args=[]):
         """
 
         if setup is not None:
-            self.env = dict(os.environ)
+            self.env = os.environ
             execute('source {0} {1}'.format(setup, " ".join(setup_args)), shell=True, env=self.env, update_env=True)
 
         else:
-            # bug #44334: Ganga/Utility/Shell.py does not save environ
-            env = dict(os.environ)
-            self.env = expand_vars(env)
+            self.env = os.environ
```

Comment:
Seems like the `self.env = os.environ` can just be put up before the if clause and then just have the `execute` part inside the if clause.